### ISSUE TITLE
FR wmllint – disable unbalanced check inside [lua] tag

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#4380]
It probably makes never sense for wmllint to look what is inside Lua.
One example: https://github.com/wesnoth/wesnoth/commit/f17119815003783f1dfe7038a7f5525f9653f103

**Comment by ProditorMagnus** [comment]
That code is balanced though?

**Comment by sevu** [comment]
Without the magic comment it reports:
```
"data/campaigns/The_Hammer_of_Thursagan/scenarios/05_Fear.cfg", line 413: unbalanced [loc_i] closed with [/lua].
"data/campaigns/The_Hammer_of_Thursagan/scenarios/05_Fear.cfg", line 714: unbalanced [loc_i] closed with [/scenario].
"data/campaigns/The_Hammer_of_Thursagan/scenarios/05_Fear.cfg", line 715: tag stack nonempty ([('scenario', {'id': '05_Fear', 'name': '_ "Fear"','next_scenario': '06_Forbidden_Forest', 'victory_when_enemies_defeated': 'no'}), ('lua', {'code': '<<', 'helper': 'wesnoth.require "helper"', 'wml_actions': 'wesnoth.wml_actions', 'vars': 'wml.variables', 'orc_types': 'vars.orc_types', 'radius': 'radius + 1', 'locs': 'wesnoth.get_locations({T["not"] { T.filter {} } , T["and"] { x = x, y = y, radius = radius } })', 'orc': 'helper.rand(orc_types)', 'loc_i': 'helper.rand("1.."..'}), ('loc_i', {}), ('loc_i', {'type': 'orc, side = "4"})'}), ('loc_i', {}), ('loc_i', {'type': 'orc, side = "4", random_traits = "yes", generate_name= "yes", upkeep = "loyal" })', 'norcs': 'nil'})]) at end of file.
```

**Comment by ProditorMagnus** [comment]
Then the issue is that wmllint looks for tags inside attribute values, not that Lua is special.

**Comment by jostephd** [comment]
Yet another reason for wmllint to use wmlparser3.

**Comment by ProditorMagnus** [comment]
After it is parsed it is one large WML dump, which can be analysed, but not changed. And unless WML has unique structure or attribute values it is hard to find where certain line is from.

**Comment by Elvish-Hunter** [comment]
Closing because it was fixed in 2a83e78.
